### PR TITLE
Support NET. 6.0

### PR DESCRIPTION
**Pull request by zcsizmadia**
- Support NET 6
- Run tests for all supported .NET versions (https://dotnet.microsoft.com/download/dotnet)
- Remove netcoreapp2.1 (EOL: https://dotnet.microsoft.com/download/dotnet/2.1)

### CODE REVIEW

File: src/GraphQL.Harness/GraphQL.Harness.csproj
Comment (sungam3r) [inline src/GraphQL.Harness/GraphQL.Harness.csproj:4]:
```suggestion
    <TargetFrameworks>net6;net5;netcoreapp3.1</TargetFrameworks>
```

File: src/GraphQL.MicrosoftDI.Tests/GraphQL.MicrosoftDI.Tests.csproj
Comment (sungam3r) [inline src/GraphQL.MicrosoftDI.Tests/GraphQL.MicrosoftDI.Tests.csproj:5]:
```suggestion
    <TargetFramework>net6</TargetFramework>
```

single value

File: src/GraphQL.Tests/Serialization/SystemTextJson/InputsConverterTests.cs
Comment (sungam3r) [inline src/GraphQL.Tests/Serialization/SystemTextJson/InputsConverterTests.cs:18]: Why did this property need? In addition, I see that it is available in NET5.

Comment (zcsizmadia) [reply src/GraphQL.Tests/Serialization/SystemTextJson/InputsConverterTests.cs:18]: 1. IgnoreNullValues is marked as obsolete. https://source.dot.net/#System.Text.Json/System/Text/Json/Serialization/JsonSerializerOptions.cs,270
2. You are right, it is available for NET 5. Changed the ifdefs.

**Comment by zcsizmadia** [comment]
I am not sure how the LGTM build cab ne fixed. AFICT it uses a default config, which does not have NEt 6 in it. Maybe  a custom lgtm.yml is needed, which installs NET 6.0 SDK?

**Comment by zcsizmadia** [comment]
I made some changes to use net5.0 or net6.0 for frameworks, instead of net5 or net6. I neevr used net5 or net6 tokens for frameowrk, and the build seems to recognized them as  NET Framework 5

File: src/GraphQL.Tests/Serialization/SystemTextJson/InputsConverterTests.cs
Comment (sungam3r) [reply src/GraphQL.Tests/Serialization/SystemTextJson/InputsConverterTests.cs:18]: I think both properties may/should be omitted for ease of reading. `IgnoreNullValues` is `false` by default and `DefaultIgnoreCondition` is `JsonIgnoreCondition.Never` by default too so why to bother with all these ifdefs ?

**Comment by sungam3r** [comment]
LGTM is broken for now, do not pay attention, it does not affect the build.

net5 and net6 is a well-knows aliases for net5.0 and net6.0, please, revert back (without `.0`).

> the dotnet format command seems not to work in the "Check formatting" gitflow step. "dotnet format --check --fix-whitespace ..." seems to be incorrect command line syntax.

This tool has been changed a bit. See here https://github.com/dotnet/format .

File: .github/workflows/codeql-analysis.yml
Comment (sungam3r) [inline .github/workflows/codeql-analysis.yml:24]: It's better to deal with codeql in a separate PR.

Comment (zcsizmadia) [reply .github/workflows/codeql-analysis.yml:24]:
```
dotnet tool install -g dotnet-format --version 6.0.243104 --add-source https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-tools/nuget/v3/index.json
          dotnet format --no-restore --verify-no-changes -v diag --severity warn whitespace || (echo "Run 'dotnet format' to fix whitespace issues" && exit 1)
          dotnet format --no-restore --verify-no-changes -v diag --severity warn analyzers || (echo "Run 'dotnet format' to fix analyzers issues" && exit 1)
```

This seems to be the updated execution script. I can add this into the PR if needed

Comment (sungam3r) [reply .github/workflows/codeql-analysis.yml:24]: Yes, `--verify-no-changes` is the right option now. Also it's worth to change messages  after `echo` to `Run 'dotnet format --no-restore --verify-no-changes -v diag --severity warn whitespace' to fix whitespace issues`. The same for analyzers.

File: src/GraphQL.Tests/Serialization/SystemTextJson/InputsConverterTests.cs
Comment (zcsizmadia) [reply src/GraphQL.Tests/Serialization/SystemTextJson/InputsConverterTests.cs:18]: I tried to keep the original way and add a compatible but not obsolete way :) However I think it is safe to remove those assignments, since they are assigned to the default values anyway. If you agree, I remove them not to create confusion or obsolete warnings.

File: .github/workflows/codeql-analysis.yml
Comment (zcsizmadia) [reply .github/workflows/codeql-analysis.yml:24]: How about bumping the version for dotnet format to 7.0.255502 (the latest on https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-tools/nuget/v3/index.json)? The new options mentioned above work as expected

File: src/GraphQL.Tests/Serialization/SystemTextJson/InputsConverterTests.cs
Comment (sungam3r) [reply src/GraphQL.Tests/Serialization/SystemTextJson/InputsConverterTests.cs]: OK to remove

File: .github/workflows/codeql-analysis.yml
Comment (sungam3r) [reply .github/workflows/codeql-analysis.yml:24]: Note that 6.0.243104 is the last version that supports netcoreapp3.1

Comment (sungam3r) [inline .github/workflows/codeql-analysis.yml:24]:
```suggestion
```

**Comment by sungam3r** [comment]
@zcsizmadia I reviewed all changed files. Ping @Shane32 for review. Waiting for tests.

**Review by sungam3r — APPROVED** [review]

**Comment by Shane32** [comment]
The GraphQL project is based on .net standard. So the changes here only affect testing and are rather unnecessary. 

I wouldn’t merge this PR until CodeQL and LGTM are working, or I would remove those tests. There’s no rush.

**Comment by sungam3r** [comment]
Makes sense. I was going to fix LGTM/CodeQL.

**Comment by SimonCropp** [comment]
@Shane32 @sungam3r anything i can help with here? I would like to add support for dateonly+timeonly and this PR looks like it should be merged before i attempt that

**Comment by Shane32** [comment]
@SimonCropp so far this only affects testing, as GraphQL.NET targets .NET Standard 2.0/2.1, neither of which support DateOnly. 

What are your intentions ?

**Comment by Shane32** [comment]
Are you thinking of adding new “built-in” graph types for DateOnly (and/or TimeOnly) to GraphQL.NET along with a net6 tfm and conditional compilation?

**Comment by SimonCropp** [comment]
> Are you thinking of adding new “built-in” graph types for DateOnly (and/or TimeOnly) to GraphQL.NET along with a net6 tfm and conditional compilation?

yep

**Comment by SimonCropp** [comment]
given this pr adds test targets for net6, i would like this merged before i work in my change

**Comment by Shane32** [comment]
Makes sense. Let’s see if @sungam3r can get lgtm and codeql working before merging.

**Comment by SimonCropp** [comment]
@Shane32 thanks

**Comment by sungam3r** [comment]
Can be merged after #2717. Also see #2718.

**Pull request merged**
